src/data/set/intervals/proj_Icc.lean

```diff
@@ -118,11 +194,37 @@ end set
 
 open set
 
-variables [preorder β] {a b : α} (h : a ≤ b) {f : Icc a b → β}
+variables [preorder β] {s t : set α} {a b : α} (h : a ≤ b) {f : Icc a b → β}
+
+protected lemma monotone.Ici_extend {f : Ici a → β} (hf : monotone f) : monotone (Ici_extend f) :=
```

Comment:
should these sorts of lemmas be tagged @[mono]?

Reply:
Can `@[mono]` deal with lemmas that have assumptions?

Reply:
Really unsure..;

Reply:
Did you try it in some example?